security/pkg/nodeagent/caclient/providers/google/client.go

```diff
@@ -134,6 +135,10 @@ func (cl *googleCAClient) getTLSDialOption() (grpc.DialOption, error) {
 }
 
 func parseZone(clusterURL string) string {
```

Comment:
https://github.com/istio/istio/blob/fd812a2700604280fbf85aab43a1f0e61716b10b/security/pkg/nodeagent/caclient/providers/google/client.go#L103

this is where this is invoked. how will return "" work for the caller there? will that be okay?

Reply:
It looks easier to do unit test by putting logic within parseZone.

Reply:
yeah, i think it should be fine to return "", and the caller has check against empty. but maybe we could consider to not use IdP exchange URL to infer to location, so the whole `parseZone` can be removed.

Reply:
Will check this in the coming sprint.